src/irmin-pack/inode.ml

```diff
@@ -616,6 +616,42 @@ struct
             add (Lazy.force t.hash) (to_bin t)
       in
       aux ~depth:0 t
+
+    let check_stable ~find t =
+      let rec check t any_stable_ancestor =
+        let stable = t.stable || any_stable_ancestor in
+        match t.v with
+        | Values _ -> true
+        | Tree tree ->
+            Array.for_all
+              (function
+                | None -> true
+                | Some t ->
+                    let t = get_target ~find t in
+                    (if stable then not t.stable else true) && check t stable)
+              tree.entries
+      in
+      check t t.stable
+
+    let contains_empty_map ~find t =
```

Comment:
The name of this function is ambiguous. We are going to return false when length=0.